### DISCUSSION TITLE
How to stream the result without waiting for the full answer

### DISCUSSION THREAD

**Discussion by OkobaPatino** [duckdb/duckdb#13384] [Q&A] [answered]
I want to select all the rows from a table using the C API, and I also want to stream them, meaning getting them one by one or chunk size (2048), but when I even use prepared statement, and pending result, the duckdb_pending_execute_task will return DUCKDB_PENDING_RESULT_NOT_READY until it allocates enough memory for all the rows (10 gigabytes for millions of rows), and then it returns DUCKDB_PENDING_RESULT_READY.
Is there a way to get the result back as soon as a chunk is ready and prevent this slow and memory intensive situation?

**Comment by Tishj** [comment, marked as answer]
You want to use a stream query result, see `duckdb_execute_prepared_streaming`

**Reply by OkobaPatino** [reply]
I saw it but it is marked as [deprecated](https://duckdb.org/docs/api/c/api#duckdb_execute_prepared_streaming). Also streaming for pending too is marked as [deprecated](https://duckdb.org/docs/api/c/api#duckdb_pending_prepared_streaming).

**Reply by Tishj** [reply]
Deprecated in this case means there will be an alternative in the future, but there is no alternative currently
In the future the same functionality will be available through a different method

**Reply by OkobaPatino** [reply]
Thank you. Is there an issue for this so I can follow for future?

**Reply by Tishj** [reply]
Not really, the best candidate would probably be the PR adding the deprecation notice, <https://github.com/duckdb/duckdb/pull/12259>
